edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Core.Test/metrics/ExceptionCounterTest.cs

```diff
@@ -132,23 +133,33 @@ public void DisposeUnsubscribes()
         {
             /* setup */
             (Dictionary<string, long> result, IMetricsProvider provider) = this.MockCounter();
-            Dictionary<string, long> expected = new Dictionary<string, long>();
-            using (new ExceptionCounter(new Dictionary<Type, string>(), new HashSet<Type>(), provider))
+            Dictionary<Type, string> recognizedExceptions = new Dictionary<Type, string>
             {
-                this.ThrowAndCatch(new Exception());
-                expected["other"] = 1;
-                Assert.Equal(expected, result);
+                { typeof(TestException1), "test1" },
+                { typeof(TestException2), "test2" },
+            };
+            using (var counter = new ExceptionCounter(recognizedExceptions, new HashSet<Type>(), provider))
+            {
+                counter.Start();
+
+                this.ThrowAndCatch(new TestException1());
+                Assert.Equal(1, result["test1"]);
             }
 
             this.ThrowAndCatch(new Exception());
-            Assert.Equal(expected, result);
+            Assert.Equal(1, result["test1"]);
```

Comment:
You should verify result.count is equal to 1 as well; or do the same verification as before.

Reply:
the reason for this change is that result wasn't equal to expected when ran in parallel. Other tests would throw exceptions, which increments the "other" value. This is intended behavior, but impossible to test adjacent to other tests.